### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-notification

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2058]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-notification component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-notification can be found here: [sd-notification Notes](https://www.notion.so/matuzo/sd-notification-1baf3f6e1bc081888699c9de9df9eeae).

------------------------------------------

## Comment
**Problem:** Announcements are inconsistent, insufficient, and confusing.
- VoiceOver on macOs announces "Close"
- Talkback on Android announces "lorem ispum close".
- NVDA doesn't announce anything except when another message is already present; then it announces "notification, lorem ipsum close button" for the new message.
- JAWS announces, "notification, lorem ipsum close"

Screen reader users either don't receive the message or get a message that includes the button label, which some users could interpret as an action they must take.

One of the reasons for the inconsistency is that with some screen readers, live regions added to the page after the page has loaded are sometimes ignored. Another is that live regions don't announce roles, only text.

**Solution 1:** Use an empty wrapper div with `role=status` that's always present for your messages. Notifications that appear inside the div will be read out. Avoid interactive elements inside your notifications.
**Solution 2:** Don't use toasts.

**Problem:** People who use software to zoom into parts on the screen may miss toast messages.
**Solution 1:** Don't use toasts for critical information. Always show critical notifications inline on the page.
**Solution 2:** Don't use toasts.

**Problem:** When the page is zoomed in, toasts may obscure large parts of the screen.
**Solution 1:** Don't use toasts and show notifications only inline.

**Problem:** Keyboard users must tab through the remaining page to close toasts or use other controls within.
**Solution 1:** Don't move focus, don't use toasts for critical information and actions, and try to avoid using interactive elements in toasts.
**Solution 2:** Don't use toasts.

**Problem:** Focus is trapped inside notifications. The only way to get out is to close notifications.
**Solution:** Don’t trap focus.

## Notes
- If someone were to ignore, or miss a toast message due to its timed display, there should be no negative impact on their current activities or the status that the message conveyed. Using the previous examples, ignoring a toast message would still mean that a file was saved, that a message was sent, or that a meeting was about to start. (Scott O’Hara)
- Consider not using buttons in timed toasts. [[Scott O’Hara explains why](https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html#inclusive-ux-of-a-toast)](https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html#inclusive-ux-of-a-toast).

## Best Practices
- Notification shouldn’t close earlier than 5 seconds.
- Don’t use toasts for critical information.
- Prefer default (inline) notifications over toast notifications.

## Accessibility information
- *Trap focus until an action is taken or the notification is dismissed.*
**Why?**
- Avoid toast messages with interactive elements inside, like buttons or links, because toast messages are hard to reach for keyboard and screen reader users.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#timing-adjustable
- https://www.w3.org/WAI/WCAG22/quickref/#reflow
- https://www.w3.org/WAI/WCAG22/quickref/#resize-text
- https://www.w3.org/WAI/WCAG22/quickref/#status-messages

## Additional resources
- https://adrianroselli.com/2020/01/defining-toast-messages.html
- https://adamsilver.io/blog/the-problem-with-toast-messages-and-what-to-do-instead/
- https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by paulovareiro29** [comment]
@coraliefeil some of these problems are something in which we don't have to do anything.
> Problem: People who use software to zoom into parts on the screen may miss toast messages.
> Solution 1: Don't use toasts for critical information. Always show critical notifications inline on the page.
> Solution 2: Don't use toasts.

> Problem: When the page is zoomed in, toasts may obscure large parts of the screen.
> Solution 1: Don't use toasts and show notifications only inline.

I believe this should be added to the documentation, if not already, so our users are aware of these a11y concerns.

Also, in our Figma documentation we have the following:
> Trap focus until an action is taken or the notification is dismissed.

While Manuel states the opposite:
> Problem: Focus is trapped inside notifications. The only way to get out is to close notifications.
> Solution: Don’t trap focus.

Which one should we take as accurate? 😄

**Comment by coraliefeil** [comment]
I’ll ask Damaris

**Comment by coraliefeil** [comment]
Pls remove focus trap in toast notifications:
https://solid-design-system.fe.union-investment.de/docs/?path=/docs/components-sd-notification--docs#toast-notification
https://solid-design-system.fe.union-investment.de/docs/?path=/docs/templates-notification--docs#info-toast-notification

Pls remove focus trap in this inline notification:
https://solid-design-system.fe.union-investment.de/docs/?path=/docs/templates-notification--docs#error--notification

I’ve [updated the docs](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=12838-17792&t=t7u95xl2TonNh9bW-11) here but have 2 comments I need your opinion on.

We’ll keep interactive elements also in toasts. It is aa1y-wise ok to do so if it is an additional/supplementary action.
Like e.g. "undo" action in a toast to delete an email ... which can also be undone from the trash folder of the email inbox.
